src/Tizen.NUI/src/public/Transition/SlideTransition.cs

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright(c) 2021 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+namespace Tizen.NUI
+{
+    using System;
+    using System.ComponentModel;
+    using Tizen.NUI.BaseComponents;
+
+    /// <summary>
+    /// SlideTransition class is a cluster of properties for the slide transition of a View.
+    /// SlideTransition provides smoothly appearing/disappearing effects for target Control.
+    /// The direction the target Control is comming from or going to can be selected in the pre-defined directions. {UP, DOWN, LEFT, RIGHT}
+    /// And, to use custom direction, the direction can be set CUSTOM and use SetDirection(Vector2) method.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class SlideTransition : TransitionBase
+    {
+        /// <summary>
+        /// Pre-defined Directions to define where the control is comming from or going to.
+        /// </summary>
+        /// <remaks>
+        /// Direction::CUSTOM provides the way to use arbitrary direction for SlideTransition.
+        /// CustomDirection property can be used to set specific direction.
+        /// If the direction is not specified, the direction is randomly selected among UP, DOWN, LEFT and RIGHT.
+        /// </remaks>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public enum SlideDirection
+        {
+            /// <summary>
+            /// View will be appeared from/disappeared to TOP of the screen.
+            /// </summary>
+            TOP,
```

Comment:
In NUI, Pascal Casing is used for the enum identifiers. (https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions#capitalization-rules-for-identifiers)

Reply:
Thank you for your check.
I changed it to "Top" at the latest patch.
And also added all the other properties with the manner.
Please review this patch once more.